include/oneapi/dpl/pstl/utils.h

```diff
@@ -41,6 +41,8 @@
 #    include <cstring> // memcpy
 #endif
 
+#define _ONEDPL_DEFINED_AND_NOT_ZERO(X) (X + 0)
```

Comment:
I think this doesn't do what we are advertising by the name (and we want the semantics of the name I think).  
`_PSTL_UDS_PRESENT` is defined to empty in the external std lib def when it is "true" and left undefined when it is "false".  

This macro will convert that empty string to a "false" rather than a "true".

I think we want it to be "true" / "1" any time it is defined and not explicitly set to "0".

Reply:
Yes, you are correct. This macro should be truthy if the input is defined and empty, but that is not the case here.

I made this [godbolt](https://godbolt.org/z/z1xK4Gd7c) to try to write a macro that has all of the properties that we want, but I cannot come up with a macro that satisfies everything. So perhaps the original version of the PR is the solution that we should go with for now.

Reply:
Yeah, this is kind of an annoying case.

How about this:
https://godbolt.org/z/GY1qTjb5e

It would require basically converting the public pstl defines to our own local copy of the pstl defines (which would be "0" or "1"), then we could use our own copy as "normal".  

Its possible that we could work it out so that our local `_ONEDPL_UDS_PRESENT` is our "local" copy, where we only attempt to define it ourselves if it is undefined.

Reply:
After some offline discussion, @danhoeflinger and I arrived at a version of the macro that now correctly determines if the input define is either 1 or defined but empty.